drivers/ieee802154/ieee802154_kw41z.c

```diff
@@ -425,25 +638,38 @@ static int kw41z_tx(struct device *dev, struct net_pkt *pkt,
 	/* Clear all IRQ flags */
 	ZLL->IRQSTS = ZLL->IRQSTS;
 
-	tx_timeout = kw41z->tx_warmup_time + KW41Z_SHR_PHY_TIME +
-		     payload_len * KW41Z_PER_BYTE_TIME;
+	/*
+	 * Current Zephyr 802.15.4 stack doesn't support ACK offload
+	 */
 
+#if KW41Z_AUTOACK_ENABLED
 	/* Perform automatic reception of ACK frame, if required */
-	if (KW41Z_AUTOACK_ENABLED) {
-		tx_timeout += KW41Z_ACK_WAIT_TIME;
+	if (KW41Z_AUTOACK_ENABLED && payload->fc.ar) {
+		tx_timeout = kw41z->tx_warmup_time + KW41Z_SHR_PHY_TIME +
+				 payload_len * KW41Z_PER_BYTE_TIME + 10 +
+				 KW41Z_ACK_WAIT_TIME;
+
+		SYS_LOG_DBG("AUTOACK_ENABLED: len: %d, timeout: %d, seq: %d",
+			payload_len, tx_timeout, payload->sequence);
```

Comment:
btw, and I noticed it in that commit but also in the following one: parameter alignment is wrong.
It should be below the last '(' + 1 space.

Your editor should be able to do it automatically for you (it needs to follow Linux code style).

Reply:
We might consider documenting this requirement somewhere and catching it with checkpatch. I've not used Linux coding style on my previous stuff, instead relying on AStyle to fix up the style based on the project requirements. On the Zephyr contribution page it lists Linux coding style as a requirement with a link to those rules and when you go to the rules the only comment about breaks in long lines is "substantially to the right" without a reference to the alignment being requested here. Maybe there is more guidance on this somewhere else or folks just get used to the specifics that are not called out. I haven't been able to figure out how to get this to occur with my editor and looking at this file I see that line breaks like this are numerous in this file. So are you wanting all of them changed?